src/dials/algorithms/indexing/indexer.py

```diff
@@ -540,7 +540,7 @@ def index(self):
                     )
                     break
 
-            n_lattices_previous_cycle = len(experiments)
+            n_lattices_previous_cycle = len(experiments.crystals())
```

Comment:
Looking at where `n_lattices_previous_cycle` is used, should L577 also be updated to `elif len(experiments.crystals())` ?